### PR TITLE
doc: Remove unicode characters, update build to assume utf-8 locale

### PR DESCRIPTION
**Pull request by chu11**
Per discussion in #3071.  Not sure if others want to add other fixes on top of this one (such as #3076).

Please take note of my removal of unicode chars changes.  Some are obvious (unicode  `…` to plain old ascii `...`).  I replaced a "-- in other words" with an "i.e." and some `"or"` with `OR` instead of just reverting to what was in the original asciidoc files.

### CODE REVIEW

**Comment by chu11** [comment]
@Mergifyio rebase

File: doc/man3/flux_fd_watcher_create.rst [hunk 1]
```diff
@@ -36,7 +36,7 @@ to monitor for events on a file descriptor *fd*. When events occur,
 the user-supplied *callback* is invoked.
 
 The *events* and *revents* arguments are a bitmask containing a logical
-“or” of the following bits. If a bit is set in *events*, it indicates
+OR the following bits. If a bit is set in *events*, it indicates
```
Comment (grondo) [inline doc/man3/flux_fd_watcher_create.rst:39]: The wording no longer makes sense as it now says "a logical OR the following bits". Maybe the `of` was inadvertently dropped?

Comment (chu11) [reply doc/man3/flux_fd_watcher_create.rst:39]: doh! thanks.

**Comment by chu11** [comment]
re-pushed w/ @grondo's find above

**Comment by chu11** [comment]
> I also pushed a commit to disable "smartquotes". (Might want to test again in your environment and ensure that doesn't break anything, just in case)

Builds for me!  Someone else will have to approve since @grondo and I both touched this

**Comment by grondo** [comment]
@Mergifyio rebase

File: doc/man1/flux-kvs.rst [hunk 2]
```diff
@@ -48,18 +48,18 @@ arguments are described below.
...
 ========
 
-**namespace create** [-o owner] *name* [*name…​*]
+**namespace create** [-o owner] *name* [*name* ...]
    Create a new kvs namespace. User may specify an alternate userid of a
    user that owns the namespace via *-o*. Specifying an alternate owner
    would allow a non-instance owner to read/write to a namespace.
 
-**namespace remove** *name* [*name…​*]
+**namespace remove** *name* [*name...*]
```
Comment (watson6282) [inline doc/man1/flux-kvs.rst:56]: You are missing a space before the "...".

Comment (grondo) [reply doc/man1/flux-kvs.rst:56]: Thanks! This PR was only removing some non-ascii characters that crept in due to auto-conversion from asciidoc to reStructuredText, and as such preserves the previous formatting I believe. If you have time, please open an issue on manpage formatting errors (I'm sure there are plenty!). We also accept pull requests! :wink:

**Comment by grondo** [comment]
@mergifyio rebase

**Pull request merged**